detectors/aws/EksDetector.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+declare(strict_types=1);
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Detectors\Aws;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Psr7\Request;
+use OpenTelemetry\Sdk\Resource\ResourceConstants;
+use OpenTelemetry\Sdk\Resource\ResourceInfo;
+use OpenTelemetry\Sdk\Trace\Attributes;
+
+/**
+ * The AwsEksDetector can be used to detect if a process is running in AWS
+ * Elastic Kubernetes and return a {@link Resource} populated with data about the Kubernetes
+ * plugins of AWS X-Ray. Returns an empty Resource if detection fails.
+ */
+class EksDetector
+{
+    // Credentials and path for locating API
+    private const K8S_SVC_URL = 'kubernetes.default.svc';
+    private const K8S_CERT_PATH = '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt';
+
+    // Path for Amazon CloudWatch Logs
+    private const AUTH_CONFIGMAP_PATH = '/api/v1/namespaces/kube-system/configmaps/aws-auth';
+    private const CW_CONFIGMAP_PATH = '/api/v1/namespaces/amazon-cloudwatch/configmaps/cluster-info';
+
+    private const CONTAINER_ID_LENGTH = 64;
+    
+    public function __construct(EksProcessDataProvider $processData, Client $guzzle)
+    {
+        $this->processData = $processData;
+        $this->guzzle = $guzzle;
+    }
+    
+    public function detect(): ResourceInfo
+    {
+        try {
+            if (!$this->isEks() || !$this->processData->isK8s()) {
```

Comment:
```suggestion
            if (!$this->processData->isK8s() || !$this->isEks()) {
```

Logically, it can't be EKS if it isn't k8s.  The k8s check is also cheaper, so do that first and short circuit if it isn't k8s.